lib/list.g

```diff
@@ -392,53 +419,32 @@ DeclareRepresentationKernel( "IsRangeRep",
 ##
 ##  <Description>
 ##  For some lists the &GAP; kernel knows that they are in fact ranges.
-##  Those lists are represented internally in a compact way instead of the
-##  ordinary way.
+##  Those lists are represented internally in a compact way,
+##  namely in <Ref Filt="IsRangeRep"/>, instead of as plain lists.
+##  A list that is represented as a plain list might still be a
+##  range but &GAP; may not know this.
 ##  <P/>
 ##  If <A>list</A> is a range then <Ref Func="ConvertToRangeRep"/> changes
-##  the representation of <A>list</A> to this compact representation.
-##  <P/>
-##  This is important since this representation needs only 12 bytes for
-##  the entire range while the ordinary representation needs <M>4 length</M>
-##  bytes.
-##  <P/>
-##  Note that a list that is represented in the ordinary way might still be a
-##  range.
-##  It is just that &GAP; does not know this.
-##  The following rules tell you under which circumstances a range is
-##  represented  in the compact way,
-##  so you can write your program in such a way that you make best use of
-##  this compact representation for ranges.
-##  <P/>
-##  Lists created by the syntactic construct
-##  <C>[ <A>first</A>, <A>second</A>  .. <A>last</A> ]</C> are of course
-##  known to be ranges and are represented in the compact way.
-##  <P/>
-##  If you call <Ref Func="ConvertToRangeRep"/> for a list represented the
-##  ordinary way that is indeed a range,
-##  the representation is changed from the ordinary to the compact
-##  representation.
+##  the representation of <A>list</A> to <Ref Filt="IsRangeRep"/>.
 ##  A call of <Ref Func="ConvertToRangeRep"/> for a list that is not a range
 ##  is ignored.
 ##  <P/>
-##  If you change a mutable range that is represented in the compact way,
-##  by assignment, <Ref Oper="Add"/> or <Ref Oper="Append"/>,
-##  the range will be converted to the ordinary representation, even if the
-##  change is such that the resulting list is still a proper range.
+##  Note that in general it is safer to directly construct <A>list</A> as a
```

Comment:
In which way is it "safer"?

Reply:
Do you mean I should add more explanation or that I should drop the comment?

I mean in the sense of:
If `list` is also known to another function `f` then e.g. `f` could afterwards assign to an entry of `list` and thus convert it back into a plain list. Or, `f` could assume that `list` is a plain list and call special kernel functions. In this case GAP could crash or - even worse - produce wrong results.

What do you think?

Reply:
I meant the question directly like that: It was completely unclear to me what you meant with "safer" here. And frankly, I am still not sure I follow after your explanation. How do you think one can trigger that crash here, from within GAP (of course from kernel code, it is always trivial to crash).

So, I'd either remove, replace, or explain this sentence, but it's not OK in the current form.

Reply:
In anything with isn't kernel documentation, then users shouldn't really worry about the representation as much as possible, and certainly shouldn't think anything is "safer".

Reply:
> So, I'd either remove, replace, or explain this sentence, but it's not OK in the current form.

I agree. In hindsight implying that `ConvertToRangeRep` is in some way "unsafe" is way over the top.

My thought was to mention that (at least from my POV) one should avoid converting or changing objects that you didn't create yourself or that other functions may depend on. But the documentation of `ConvertToRangeRep` is hardly the place to do that. I'm gonna remove this sentence from my PR. Just a thought: maybe a comment along these lines may be more appropriate, _if at all_, in the documentation of `MakeImmutable`.

Reply:
I removed the sentence.